.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,37 @@
+name: PR Checks
+
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, reopened, edited, ready_for_review]
+
+jobs:
+  run-pr-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: PR description check
+        env:
+          DESCRIPTION: ${{ github.event.pull_request.body }}
+        run: |
+          if [[ $DESCRIPTION == *"Describe what you did and why"* ]]; then
+            echo "Please add a meaningful description to your PR."
+            exit 1
+          fi
+      - name: PR commit count check
+        env:
+          COMMIT_COUNT: ${{ github.event.pull_request.commits }}
+        run: |
+          commit_limit=100
```

Comment:
100 commits on a single PR seems to be very high, should we keep it as 20 or something lower?

Reply:
Completely agree, but then again 20 might cause some issues. I would be up for 20 commits for sure but not sure if everyone is on the same page.

Reply:
I agree 100 is very high. I can't see many situations where more than 20 would be needed but that doesn't mean they don't reasonably exist. Could be a split the difference area? Like 40? Or otherwise keep as is and it's always something we can review and bring down in the future? 

Or go the other way and bring it down and if we hit the limit and realise it's too low, it's not a huge change to up it and review with the team. 

Just some thoughts. Happy with these changes though :)

Reply:
Maybe I'll bring it down to 40 then? If we need to bump it up (which we really shouldn't) then we can.